src/Subject.ts

```diff
@@ -2,236 +2,136 @@ import {Operator} from './Operator';
 import {Observer} from './Observer';
 import {Observable} from './Observable';
 import {Subscriber} from './Subscriber';
-import {Subscription, ISubscription, TeardownLogic} from './Subscription';
-import {SubjectSubscription} from './SubjectSubscription';
-import {$$rxSubscriber} from './symbol/rxSubscriber';
-
-import {throwError} from './util/throwError';
+import {ISubscription, Subscription} from './Subscription';
 import {ObjectUnsubscribedError} from './util/ObjectUnsubscribedError';
+import {SubjectSubscription} from './SubjectSubscription';
 
 /**
  * @class Subject<T>
  */
-export class Subject<T> extends Observable<T> implements Observer<T>, ISubscription {
+export class Subject<T> extends Observable<T> implements ISubscription {
+  observers: Observer<T>[] = [];
 
-  static create: Function = <T>(destination: Observer<T>, source: Observable<T>): Subject<T> => {
-    return new Subject<T>(destination, source);
-  };
+  isUnsubscribed = false;
 
-  constructor(protected destination?: Observer<T>, protected source?: Observable<T>) {
+  isStopped = false;
```

Comment:
Just an observation the lack of `private`/`protected`/`public` qualifiers makes all these fields essentially public to TypeScript consumers, some are useful (`hasError` for example) others may not be so useful.